lib/kvdb/ikvdb.c

```diff
@@ -1024,7 +1024,7 @@ ikvdb_alloc(
 
     memset(self, 0, sz);
     self->ikdb_rp = *params;
-    self->ikdb_read_only = params->read_only;
+    self->ikdb_read_only = (params->mode != KVDB_MODE_RDWR);
```

Comment:
Do we still need this struct member?

Reply:
Yes, it is convenient to operate with just a flag at the ikvdb layer and below that indicates whether the DB is read-only or not.